.github/workflows/naughty-prune.yml

```diff
@@ -1,20 +1,31 @@
 name: prune naughties
 on:
   schedule:
-    - cron: '30 1 * * *'
+    - cron: '30 1 * * 0'
   # can be run manually on https://github.com/cockpit-project/bots/actions
   workflow_dispatch:
 jobs:
   maintenance:
     runs-on: ubuntu-latest
+    environment: self
+    permissions:
+      issues: read
+      pull-requests: write
+      statuses: write
     steps:
       - name: Set up secrets
         run: echo '${{ secrets.GITHUB_TOKEN }}' > ~/.config/github-token
 
       - name: Clone repository
         uses: actions/checkout@v2
         with:
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
```

Comment:
That's not gonna be enough.  The problem is that the action that would trigger the bots to run in this case is not the pushing of the branch, but the opening of the PR.  That's still done with the GitHub API via the GitHub Actions token.

This is why we need to do a separate force-push step in the `npm-update` workflow.

This is all terribly inconvenient...

Reply:
Well, it is demonstrably enough -- PR #3364 has all tests, and it gets force-pushed so that the bot can add its `Closes #` line.

Reply:
BTW, this is exactly the same why https://github.com/cockpit-project/cockpit/blob/main/.github/workflows/weblate-sync-po.yml works. The point of the ssh git push is not for the initial branch creation, but the force-push for the "closes #" which both po-refresh and naughty-prune do (via common code in tasks/).